### PR TITLE
Introduce Sequence flags

### PR DESCRIPTION
**Pull request by engineer124**
I also gave some of these flags names where there was enough documentation to work out their purpose

### CODE REVIEW

**Review by Roman971 — APPROVED** [review]

File: src/code/code_800EC960.c [hunk 1]
```diff
@@ -143,117 +143,127 @@ u8 sAudioExtraFilter2 = 0;
 Vec3f* sSariaBgmPtr = NULL;
 f32 D_80130650 = 2000.0f;
 u8 sSeqModeInput = 0;
+
+#define SEQ_FLAG_ENEMY (1 << 0)
+#define SEQ_FLAG_FANFARE (1 << 1)
+#define SEQ_FLAG_FANFARE_GANON (1 << 2)
+#define SEQ_FLAG_RESTORE (1 << 3)
```
Comment (Dragorn421) [inline src/code/code_800EC960.c:150]: I think just "restore" is unclear about "can this sequence be restored" (it isn't afaict) vs "can a sequence be restored when the sequence is stopped"

Actually I think looking at `func_800F5B58` that if this flag isn't set the sequence isn't even stopped? maybe these can be called "temporary playing sequences" ? :thinking: it's weird that specific functions exist but check this flag nontheless

File: src/code/code_800EC960.c [hunk 2]
```diff
@@ -143,117 +143,127 @@ u8 sAudioExtraFilter2 = 0;
...
+#define SEQ_FLAG_4 (1 << 4)
+#define SEQ_FLAG_5 (1 << 5)
+#define SEQ_FLAG_6 (1 << 6)
+#define SEQ_FLAG_DISABLE_AMBIENCE (1 << 7)
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/code/code_800EC960.c:154]: No, ambience is very specific to sequence 1, you'll only find the ambience sequence there.

The thing is that ambience plays on `BGM_MAIN`, which many scene sequences also play on `BGM_MAIN`. So there is a swap between a scene sequence and ambience (either through day/night or playing SoS). So having this flag means that you can't swap out that sequence for an ambience due to night-time/SoS

File: src/code/code_800EC960.c [hunk 3]
```diff
@@ -143,117 +143,127 @@ u8 sAudioExtraFilter2 = 0;
 Vec3f* sSariaBgmPtr = NULL;
 f32 D_80130650 = 2000.0f;
 u8 sSeqModeInput = 0;
+
+#define SEQ_FLAG_ENEMY (1 << 0)
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/code/code_800EC960.c:147]: Are names ever final? Especially with this file that still needs so much work. I like `SEQ_FLAG_ALLOW_ENEMY_BGM` tbh. And the enemy bgm system is inherently designed for main-sub bgm. You could in theory play any sequence on the main-bgm and this flag would apply. That seems more like a finer detail for comments

File: src/code/code_800EC960.c (hunk 1 above)
Comment (engineer124) [reply src/code/code_800EC960.c:150]: Yeah, it restores but under a specific set of conditions with a specific function, so I don't really know how to describe that with a short name, so I kept it simple and would explain it more with comments later (I intend to properly comment and adjust these flags later, this is indeed a first draft)

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**